google-cloud-notification/src/main/java/com/google/cloud/notification/Notification.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.notification;
+
+import com.google.api.core.BetaApi;
+import com.google.cloud.Service;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.util.List;
+
+/**
+ * An interface for notifications in Google Cloud Storage.
+ *
+ * @see <a href="https://cloud.google.com/storage/docs">Google Cloud Storage</a>
+ */
+public interface Notification extends Service<StorageOptions> {
```

Comment:
If storage already has notification methods, why do we need a separate class for it?

Reply:
Because there's some entities in Pubsub that I want to use in the Notification module. Specifically:
- TopicName from Pubsub, to make it easy to construct a topic name instead of requiring the user to correctly format `projects/{project}/topics/{topic}`.
- For system tests, Pubsub is required to make the Topic before Storage can make a Notification on that Topic.

I talked to Garrett, and he suggested making a separate module that depends on both google-cloud-storage and google-cloud-pubsub instead of requiring google-cloud-storage to depend on google-cloud-pubsub.

Reply:
I see. I missed the fact that the feature is only added to storage RPC, not the "friendly" storage. Makes sense now.